lib/puppet/pops/parser/egrammar.ra

```diff
@@ -344,7 +344,12 @@ if_expression
   # Produces Model::IfExpression
   if_part
     : expression LBRACE statements RBRACE else {
-        result = Factory.IF(val[0], Factory.block_or_expression(val[2]), val[4])
+        then_result = Factory.block_or_expression(val[2])
+        if val[2].size > 1
+          loc(then_result, val[1], val[3])
+        end
```

Comment:
I think it is best to change `Factory#block_or_expression` to make it take `from` and `to` locations - this because all calls to `block_or_expression` should provide location information for the block that it creates. 
This is better also because this second check that the array of expressions > 1 may in the future not be the only decision that `block_or_expression` makes - it could expand a single thing, or optimize multiple into one (it does neither today) - it also makes the code more DRY.